src/utils/common.ts

```diff
@@ -129,10 +137,137 @@ export const dirtyFieldsToRequestObject = (dirtyFields: object, form: IAppForms)
 
   syncPlanningDates(request, form);
   syncConstructionDates(request, form);
+  syncFinances(request, project);
 
   return request;
 };
 
+/**
+ * Make sure the finances are moved to correct years if dates are changed
+ * @param project IProject being updated
+ * @param data IProjectRequest object with data to be patched
+ */
+export const syncFinances = (request: IProjectRequest, project: IProject | null) => {
+  if (!project?.finances) {
+    return;
+  }
+
+  /**
+   * Moves the provided finance value to its new finance location and sets current to null
+   * @param newFieldYear year for the new finance field which needs to be updated
+   * @param financeValue finance value which will be summed to newFieldYear field in the budgetObject
+   * @param currentFinanceFieldName finance field which will be set to null as its value has been moved 
+   * @param budgetObject ProjectFinance object to update
+   */
+  const updateFinanceValue = (
+    newFieldYear: number,
+    financeValue: number,
+    currentFinanceFieldName: keyof IProjectFinances,
+    budgetObject: IProjectFinances,
+  ) => {
+    // Year number to finance field name mapping
+    const yearToBudgetKey: { [key: number]: keyof IProjectFinances } = {
+      0: 'budgetProposalCurrentYearPlus0',
+      1: 'budgetProposalCurrentYearPlus1',
+      2: 'budgetProposalCurrentYearPlus2',
+      3: 'preliminaryCurrentYearPlus3',
+      4: 'preliminaryCurrentYearPlus4',
+      5: 'preliminaryCurrentYearPlus5',
+      6: 'preliminaryCurrentYearPlus6',
+      7: 'preliminaryCurrentYearPlus7',
+      8: 'preliminaryCurrentYearPlus8',
+      9: 'preliminaryCurrentYearPlus9',
+      10: 'preliminaryCurrentYearPlus10',
+    };
+    const newFinanceFieldName: keyof IProjectFinances =
+      yearToBudgetKey[newFieldYear - budgetObject.year];
+    (budgetObject[newFinanceFieldName] as string) = String(
+      safeParseInt(budgetObject[newFinanceFieldName]) + financeValue,
+    );
+    (budgetObject[currentFinanceFieldName] as null) = null;
```

Comment:
Why budgetObject[currentFinanceFieldName] casted as null and set its value to null? Is intension to delete the value or just set to null?

Reply:
To set it to null, the field has to be casted as null or else it throws an error. 
Intention is to delete the finance value from that field, it is set to null to make sure it is not visible in planning view timeline.